src/Clapi/Types/Tree.hs

```diff
@@ -53,6 +53,8 @@ data TreeType
   | TtFloat (Bounds Float)
   | TtDouble (Bounds Double)
   | TtString Text
+  -- FIXME: kinda want this to be TtRef (Tagged Definition TypeName) but that
```

Comment:
This cycle is annoying, I think I got around it by defining something more abstract (I think it was the kind actually) that was basically an enumeration of the different types of type that we use, and a type family to line it up with definition later. See the branch of despair for the good and bad of this I guess.